### PR TITLE
[Bot Rules] CODEOWNERS Sync

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to capture updates to CODEOWNERS for service owners and new PR paths/labels.  Also included are some typo fixes for existing service owners.

# Related

- #26637
- #26644
- #26574

### CODE REVIEW

File: .github/fabricbot.json [hunk 1]
```diff
@@ -1059,6 +1060,25 @@
               ""
             ]
           },
+          {
+            "label": "ML-Local Endpoints",
+            "pathFilter": [
+              "sdk/ml/azure-ai-ml/azure/ai/ml/_local_endpoints/"
```
Comment (xiangyan99) [inline .github/fabricbot.json:1066]: Looks like this is a private path?

Comment (jsquire) [reply .github/fabricbot.json:1066]: It was added as part of a CODEOWNERS change in the linked PRs.    Should this not exist in either place?

Comment (xiangyan99) [reply .github/fabricbot.json:1066]: Double checking with ML team.

Comment (xiangyan99) [reply .github/fabricbot.json:1066]: We decided to discuss the issue later. :)

**Review by xiangyan99 — APPROVED** [review]

**Pull request merged**